blocks/bricks/sequence_generators.py

```diff
@@ -226,7 +238,7 @@ def generate_states(self):
     @generate.property('outputs')
     def generate_outputs(self):
         return (self.state_names + ['outputs']
-                + self.glimpse_names + ['costs'])
+                + self.glimpse_names + ['costs', 'probs'])
```

Comment:
Well, there is no sense to do beam search when the emitter is deterministic.

In fact I would like to suggest a different way of retrieving probabilities for beam search from the sequence generator. It seems to me that one can use the generation computation graph and retrieve the probabilities from there using the `VariableFilter`. Knowledge that probabilities are an output of a certain application method of a certain brick is sufficient for that. That would allow to keep the interface a bit cleaner. Check the reverse words demo for an example of using `VariableFilter` for getting intermediate results from the computation graph.